File: apps/api/bin/main/com/productivityos/project/domain/Project.kt
package com.productivityos.project.domain

import java.time.Instant
import java.time.LocalDate
import java.util.UUID

data class Project(
    val id: UUID,
    val userId: UUID,
    val title: String,
    val description: String? = null,
    val goalId: UUID? = null,
    val status: ProjectStatus = ProjectStatus.DRAFT,
    val deadline: LocalDate? = null,
    val completedAt: Instant? = null
) {
    fun activate(): Project {
        require(status == ProjectStatus.DRAFT) { "Only Draft projects can be activated" }
        return copy(status = ProjectStatus.ACTIVE)
    }

    fun returnToDraft(): Project {
        require(status == ProjectStatus.ACTIVE) { "Only Active projects can return to Draft" }
        return copy(status = ProjectStatus.DRAFT)
    }

    fun complete(now: Instant): Project {
        require(status == ProjectStatus.ACTIVE) { "Only Active projects can be completed" }
        return copy(status = ProjectStatus.COMPLETED, completedAt = now)
    }

    fun archive(): Project {
        require(status == ProjectStatus.COMPLETED) { "Only Completed projects can be archived" }
        return copy(status = ProjectStatus.ARCHIVED)
    }
}

File: apps/ios/ProductivityOS/Models/Project.swift
import Foundation

/// Project status from the backend `ProjectStatus` enum.
public enum ProjectStatus: String, Codable, CaseIterable, Sendable {
    case draft = "DRAFT"
    case active = "ACTIVE"
    case completed = "COMPLETED"
    case archived = "ARCHIVED"
}

/// Project domain model matching backend `ProjectResponse`.
public struct Project: Identifiable, Codable, Hashable, Sendable {
    public let id: UUID
    public let userId: UUID
    public let title: String
    public let description: String?
    public let goalId: UUID?
    public let status: ProjectStatus
    public let createdAt: Date?
    public let updatedAt: Date?

    public init(
        id: UUID,
        userId: UUID,
        title: String,
        description: String? = nil,
        goalId: UUID? = nil,
        status: ProjectStatus,
        createdAt: Date? = nil,
        updatedAt: Date? = nil
    ) {
        self.id = id
        self.userId = userId
        self.title = title
        self.description = description
        self.goalId = goalId
        self.status = status
        self.createdAt = createdAt
        self.updatedAt = updatedAt
    }
}
